src/herder/TxSetFrame.cpp

```diff
@@ -106,16 +123,16 @@ SeqSorter(TransactionFrameBasePtr const& tx1,
     * transactions for an account are sorted by sequence number (ascending)
     * the order between accounts is randomized
 */
-std::vector<TransactionFrameBasePtr>
-TxSetFrame::sortForApply()
+void
+TxSetFrame::sortInApplyOrder()
```

Comment:
I wonder if we really need to persist the apply order as a class member instead of just returning a properly sorted vector. What problem do we solve by this change?

Reply:
I don't see a better way if we want to get rid of the abstract class approach. 
Here in `TxSetFrame`, the transactions are guaranteed to always maintain hash order. In the "simulation" case (previously `TxSimTxSetFrame`), the transactions also need to maintain its original order as the apply order. Unless we override the `TxSetFrame` class with a different "sortInApplyOrder" behavior, this is the best way I can think of.

Reply:
Oh, right, I forgot about that. This is still a bit hacky, but should be ok. I guess an alternative would be to just make `sortForApply` virtual and have `TxSimTxSetFrame` inherited from `TxSetFrame` directly (no ABC), but I don't have a strong opinion on this.

Reply:
Fixed. Same comment as [below](https://github.com/stellar/stellar-core/pull/3395#discussion_r854371104).